sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -121,6 +122,68 @@ def test_imputation_mean_median_only_zero():
                       statistics_median, 0)
 
 
+def test_imputation_empty_column_missing_zero():
+    X = np.array([
+        [np.nan, 0, 0, 0, 5],
+        [np.nan, 1, 0, np.nan, 3],
+        [np.nan, 2, 0, 0, 0],
+        [np.nan, 6, 0, 5, 13],
+    ])
+
+    X_imputed_mean = np.array([
+        [-1, 3, -1, -1, 5],
+        [-1, 1, -1, -1, 3],
```

Comment:
You seem to be asserting that some strange behaviour regarding nans is maintained. I'd argue that behaviour -- where a feature gets considered "empty" due to the presence of a single NaN where `missing_values != 'nan'` -- is a bug.